tools/codegen/pkg/modelgen/diag.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package modelgen
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+const (
+	ERROR_DIAG DiagKind = iota
+	WARNING_DIAG
+)
+const UNKNOWN_LINE = ""
+const UNKNOWN_FILE = ""
+
+type (
+	DiagKind uint8
+
+	// Diag represents diagnostic information.
+	Diag struct {
+		kind     DiagKind
+		Location location
+		Message  string
+	}
+	// location represents the location of the Diag
+	location struct {
+		File string
+		// TODO: Currently Line is always set as UNKNOWN_LINE. Consider using proto's
```

Comment:
Agreed. I was just thinking to move till the next PR.

I have started working on it. Are you okay if I send a separate PR with that change on top of this PR (should be around couple of days after this PR is done).